File: front/vue/src/api/message.js
import api from '@/api'

// 获取未读消息数量
export function getUnreadMessageCount(id) {
    return api.get('/message/unreadCount',{
        params: {
            id: id
        }
    })
}

// 获取消息列表
export function getMessageList(page,pageSize,id) {
    return api.get('/message/list',{
        params: {
            page: page,
            pageSize: pageSize,
            id: id
        }
    })
}

// 更新消息
export function updateMessage(data){
    return api.put('/message/update',data)
}

// 删除消息
export function deleteMessage(id){
    return api.delete('/message/delete',{
        params: {
            id: id
        }
    })

}

// 一键已读所有消息
export function readAllMessage(id){
    return api.put('/message/readAll',id)
}

// 处理请求
export function handleRequest(data){
    return api.post('/message/handle',data)
}

// 添加消息
export function addMessage(data){
    return api.post('/message/add',data)
}

File: back/routers/message.go
package routers

import (
	"back/configs"
	"back/controllers"
	"back/repository"
	"back/service"
	"github.com/gin-gonic/gin"
)

func SetMessageRoutes(r *gin.Engine) {
	// 初始化依赖
	messageRepository := repository.NewMessageRepository(configs.MysqlDb)
	messageService := service.NewMessageService(messageRepository)
	messageController := controllers.NewMessageController(messageService)

	message := r.Group("/message")
	{
		// 获取用户未读消息数量
		message.GET("/unreadCount", messageController.GetUnreadMessageCount)
		// 分页查询消息
		message.GET("/list", messageController.GetMessageList)
		// 更新消息
		message.PUT("/update", messageController.UpdateMessage)
		// 删除消息
		message.DELETE("/delete", messageController.DeleteMessage)
		// 一键已读
		message.PUT("/readAll", messageController.ReadAllMessage)
		// 处理请求
		message.POST("/handle", messageController.HandleRequest)
	}
}
